src/ranking.jl

```diff
@@ -171,3 +171,162 @@ Missing values are assigned rank `missing`.
+quantile rank of x means that (x*100)% of the elements in `v` are lesser or lesser-equal the 
+given `value`. 
+
+The keyword argument `method` (default `:inc`) can be:
+
+`:inc` - The inverse of the quantile based in def. 7 in Hyndman and Fan (1996). 
+(equivalent to Excel `PERCENTRANK` and `PERCENTRANK.INC`)
+
+`:exc` - The inverse of the quantile based in def. 6 in Hyndman and Fan (1996). 
+(equivalent to Excel `PERCENTRANK.EXC`)
+
+`:compete` - Based on the `competerank` of StatsBase. 
+(equivalent to MariaDB `PERCENT_RANK`, dplyr `percent_rank`)
+
+`:tied` - Based in the def. in Roscoe, J. T. (1975). 
+(equivalent to `mean` argument of Scipy `percentileofscore`)
+
+`:strict` - Equivalent to `strict` method of Scipy `percentileofscore`.
+
+`:weak` - Equivalent to `weak` method of Scipy `percentileofscore`.
+
+!!! note
+    An `ArgumentError` is thrown if `v` contains `NaN` or `missing` values
+    and if `v` is empty or contains only one element.
+
+The function also supports types that have the mathematical properties of rings or total 
+order.
+
+# References
+[Percentile Rank on Wikipedia](https://en.wikipedia.org/wiki/Percentile_rank) covers 
+definitions and examples.
+
+Roscoe, J. T. (1975). "[Fundamental Research Statistics for the Behavioral Sciences 
+(2nd ed.)](http://www.bryanburnham.net/wp-content/uploads/2014/07/Fundamental-Statistics-
+for-the-Behavioral-Sciences-v2.0.pdf#page=57)".
+    ISBN 0-03-091934-7.
+
+Hyndman, R.J and Fan, Y. (1996) "[Sample Quantiles in Statistical Packages]
+(https://www.amherst.edu/media/view/129116/original/Sample+Quantiles.pdf)",
+    *The American Statistician*, Vol. 50, No. 4, pp. 361-365
+
+[Quantile on Wikipedia](https://en.m.wikipedia.org/wiki/Quantile) details the different 
+quantile definitions.
+
+# Examples
+```julia
+julia> using StatsBase
+
+julia> v1 = [1, 1, 1, 2, 3, 4, 8, 11, 12, 13];
+
+julia> v2 = [1, 2, 3, 5, 6, missing, 8];
+
+julia> v3 = [1, 2, 3, 4, 4, 5, 6, 7, 8, 9];
+
+# use `skipmissing` in vectors with missing entries
+julia> quantilerank(v1, 2), quantilerank(skipmissing(v2), 4)
+(0.3333333333333333, 0.5)
+```
+
+# use `Ref` to treat vector `v3` as a scalar during broadcasting.
+```julia
+julia> quantilerank.(Ref(v3), [4, 8])
+2-element Vector{Float64}:
+ 0.3333333333333333
+ 0.8888888888888888
+```
+"""
+function quantilerank(v::AbstractVector, value; method::Symbol=:inc)
+    # checks
+    value isa Number && isnan(value) &&
+        throw(ArgumentError("value cannot be NaN"))
+    any(x -> ismissing(x) || (x isa Number && isnan(x)), v) &&
+        throw(ArgumentError("vector `v` cannot contain missing or NaN entries"))
+
+    n = length(v)
+
+    n == 0 && throw(ArgumentError("vector `v` is empty. Insert a non-empty vector"))
+    n == 1 && throw(ArgumentError("vector `v` has only 1 value. Use a vector with more elements"))
+
+    count_less = count_equal   = 0
+    last_less  = first_greater = value
+    @simd for x in v
```

Comment:
I think that here `@simd` is not needed as we do branching inside anyway.

Reply:
Understood. I'll change it in the next commit.

I only used @simd because I learned that its use applies in scenarios where the for-loop has associative and commutative operations. Which seemed to be the case, but probabily I missed something. Could you please explain to me why the if-elses (branching) make the use of @simd unnecessary?
I also got to test with @btime and saw an improvement, even if small.

Reply:
My experience is that is code is branching then the compiler optimizations that `@simd` does are negligible. Maybe this is not the case for these functions.

However as the docs of `@simd` says in a warning:

> This feature is experimental and could change or disappear in future versions of Julia. Incorrect use of the `@simd` macro may cause unexpected results.

So my personal preference is not to use it unless there is a clearly visible benefit (as what we write now might get untouched for next 5 years and in the mean time the warning might become relevant).

Reply:
Also `@simd` will not work correctly with `skipmissing` so then we would have a separate function for this case only as:
```
julia> x = [1, 2, missing, 3, 4, missing]
6-element Vector{Union{Missing, Int64}}:
 1
 2
  missing
 3
 4
  missing

julia> function f(x)
       c = 0
       @simd for v in skipmissing(x)
           c += v
       end
       return c
       end
f (generic function with 1 method)

julia> f(x)
ERROR: MethodError: no method matching length(::Base.SkipMissing{Vector{Union{Missing, Int64}}})
```

Reply:
Now, I fully understand. Tks for the explanation.
Adjust done in commit bb6ba17.